framework/include/cppmicroservices/detail/TrackedService.tpp

```diff
@@ -53,6 +61,15 @@ void TrackedService<S,TTT>::ServiceChanged(const ServiceEvent& event)
     return;
   }
 
+  (void)latch.CountUp();
+  ScopeGuard sg([this]() {
+    // By using try/catch here, we ensure that this lambda function doesn't
+    // throw inside ScopeGuard's dtor.
+    try {
+      latch.CountDown();
+    } catch (...) { }
+  });
+
```

Comment:
there's still a race here; a thread can enter `void TrackedService<S,TTT>::ServiceChanged(const ServiceEvent& event)` after `void ServiceTracker<S,T>::Close()` has finished waiting on all customizers to finish.